tests/integ/modin/frame/test_unstack.py

```diff
@@ -0,0 +1,49 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+import modin.pandas as pd
+import numpy as np
+import pandas as native_pd
+import pytest
+
+from tests.integ.modin.sql_counter import sql_count_checker
+from tests.integ.modin.utils import eval_snowpark_pandas_result
+
+
+@sql_count_checker(query_count=1)
+def test_unstack_input_no_multiindex():
+    # Only 1 Query as input dataframe to unstack is not a multi-index,
+    # so calls df.reset_index().melt()
+    index = native_pd.MultiIndex.from_tuples(
+        [("one", "a"), ("one", "b"), ("two", "a"), ("two", "b")]
+    )
+    # Note we call unstack below to create a dataframe without a multiindex before
+    # calling unstack again
+    native_df = native_pd.Series(np.arange(1.0, 5.0), index=index).unstack(level=0)
```

Comment:
since unpviot have issue with missing values, do we have any test that have missing value there?

Reply:
Yes we test unpivot with missing values as part of the `df.stack()` PR: https://github.com/snowflakedb/snowpark-python/pull/1821/files. I don't think it's worth adding a duplicate of that test here since it's a very similar implementation with calling melt then unpivot (just without the `sort_rows_by_column_values`).